SmartDeviceLink/SDLTextAndGraphicManager.m

```diff
@@ -157,6 +163,10 @@ - (void)sdl_updateWithCompletionHandler:(nullable SDLTextAndGraphicUpdateComplet
         SDLLogV(@"Images already uploaded, sending full update");
         // The files to be updated are already uploaded, send the full show immediately
         self.inProgressUpdate = fullShow;
+    } else if ([self sdl_artworkUploadsFailed:self.primaryGraphic secondaryGraphic:self.secondaryGraphic maxRetryCount:MAX_ARTWORK_UPLOAD_RETRY_ATTEMPTS]) {
```

Comment:
I'm skeptical that this is the right way to do this. This is quite complicated, and putting a new `if` in here seems incorrect and that it will cause bad behavior. That's just a gut feeling. Can't this be better implemented within `sdl_uploadImagesWithCompletionHandler`? Keep a counter, if one has an error, recheck the image is still needed, and retry? If it fails, then return the handler?

Reply:
Fixed. Refactored to make it more readable.